### PR TITLE
Requester page map

### PR DESCRIPTION
**Pull request by EgorKrasno**
- Automatically gets user location, converts to MGRS and sets Location
- Added Reset Icon - resets to current user location
- Added Map Icon, brings up map that allows user to drag a marker to the HLZ location
- There is a bug that causes a crash if you close the map while it is loading, i'll fix it in next commit

### CODE REVIEW

File: src/frontend/src/component/MapModal.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            <DialogContent
+                sx={{alignContent: "center", padding: 0, margin: 0}}>
+                <Box ref={mapContainer} sx={{
+                    width: "100%",
+                    height: "100%",
+                    display: 'flex',
+                    justifyContent: 'center',
+                    alignItems: 'center'
+                }}>
+                    <CircleLoader color="#1976D2" loading={!mapLoaded} size={200}/>
```
Comment (jfarrin73) [inline src/frontend/src/component/MapModal.js:121]: MaterialUI has a circular loading indicator that we're using on the authentication loading page.

File: src/frontend/src/component/MapModal.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+} from "@mui/material";
+import mapboxgl from '!mapbox-gl'; // eslint-disable-line import/no-webpack-loader-syntax
+import CloseIcon from "@mui/icons-material/Close";
+import {forwardRef, useEffect, useRef, useState} from "react";
+import 'mapbox-gl/dist/mapbox-gl.css';
+import mgrs from "mgrs";
+import Box from "@mui/material/Box";
+import {CircleLoader} from "react-spinners";
+
+mapboxgl.accessToken = 'pk.eyJ1IjoiZWdvcmtyYSIsImEiOiJja21uMXk5OHMwdTN4Mm9wbDVpOXllcGY0In0.ZheSrkcBpR9hmpfG0qW5EQ';
```
Comment (jfarrin73) [inline src/frontend/src/component/MapModal.js:19]: should this token be an environment variable? Same question for line 34.

File: src/frontend/src/component/MapModal.js (hunk 1 above)
Comment (EgorKrasno) [reply src/frontend/src/component/MapModal.js:121]: but does it look as cool as mine?

File: src/frontend/src/component/MapModal.js (hunk 2 above)
Comment (EgorKrasno) [reply src/frontend/src/component/MapModal.js:19]: It's not a secret token, I wasn't sure if we want to offload it to an environment variable.

Comment (EgorKrasno) [reply src/frontend/src/component/MapModal.js:19]: actually, I will, it won't be added to .gitignore but I will be able to reuse it in the report map

File: src/frontend/src/component/MapModal.js (hunk 1 above)
Comment (jfarrin73) [reply src/frontend/src/component/MapModal.js:121]: nope, just throwing it out

**Pull request merged**